### ISSUE TITLE
[bug] false positive for `@eslint-react/no-unnecessary-use-prefix` rule

### ISSUE THREAD

**Issue by mcmxcdev** [Rel1cx/eslint-react#1181]
### Describe the bug

We have code like below in a file called `hooks.test.tsx`:

```ts
const mockUseSession = vi.fn().mockReturnValue(sessionUser);

vi.mock("src/components/session/session", () => ({
  useSession: () => mockUseSession(),
}));
```

The content of `src/components/session/session` is as following:
```ts
export const useSession = (): Session | undefined => {
  const { session } = React.useContext(SessionContext);

  return session;
};
```

`useSession` in `vi.mock` is getting flagged with `If your function doesn't call any Hooks, avoid the 'use' prefix. Instead, write it as a regular function without the 'use' prefix.` but I would argue that `useSession` is a hook.

If the rule is unable to determine with certainty if it's actually a hook or not, it should rather not report on it.

### Expected behavior

Possibly `eslint-react` opting out of checking `vi.mock` blocks completely.

### Platform and versions

```Text
Node.js v22.15.0

"@eslint-react/eslint-plugin": "2.0.0-beta.34"
```

### Stack trace

```Text

```

**Comment by Rel1cx** [comment]
This rule should not apply to code in test files, especially where mocks are written. IMO, a better approach is to add the following configuration in `eslint.config.ts` to opt out of inapplicable rules
```ts
{
  files: [
    "**/*.test.{ts,tsx}",
  ],
  rules: {
    // ...
    "@eslint-react/no-unnecessary-use-prefix": "off",
  },
}
```
For similar issues and discussions, please refer to https://github.com/facebook/react/issues/31237#issuecomment-2706402535

**Comment by mcmxcdev** [comment]
That is one solution, but I am wondering if there might be use cases where this rule should apply to code in test files.

**Comment by Rel1cx** [comment]
> That is one solution, but I am wondering if there might be use cases where this rule should apply to code in test files.

I will keep this issue open for a while to see if there is a change to improve this rule.
